src/openforms/js/components/admin/form_design/logic/actions/dmn/DMNParametersForm.js

```diff
@@ -0,0 +1,73 @@
+import {useFormikContext} from 'formik';
+import React, {useContext, useState} from 'react';
+import {FormattedMessage} from 'react-intl';
+import {useAsync} from 'react-use';
+
+import {FormContext} from 'components/admin/form_design/Context';
+import {DMN_DECISION_DEFINITIONS_PARAMS_LIST} from 'components/admin/form_design/constants';
+import {get} from 'utils/fetch';
+
+import VariableMapping from './VariableMapping';
+
+const DMNParametersForm = () => {
+  const {
+    values: {pluginId, decisionDefinitionId, decisionDefinitionVersion},
+  } = useFormikContext();
+  const {formVariables} = useContext(FormContext);
+
+  const [dmnParams, setDmnParams] = useState({inputs: [], outputs: []});
+
+  const {loading} = useAsync(async () => {
+    if (!pluginId || !decisionDefinitionId) {
+      setDmnParams({inputs: [], outputs: []});
+      return;
+    }
+
+    const queryParams = {
+      engine: pluginId,
+      definition: decisionDefinitionId,
+    };
+
+    if (decisionDefinitionVersion) {
+      queryParams.version = decisionDefinitionVersion;
+    }
+
+    const response = await get(DMN_DECISION_DEFINITIONS_PARAMS_LIST, queryParams);
+
+    setDmnParams({
+      inputs: response.data.inputs.map(inputParam => [inputParam.expression, inputParam.label]),
+      outputs: response.data.outputs.map(outputParam => [outputParam.name, outputParam.label]),
+    });
+  }, [pluginId, decisionDefinitionId, decisionDefinitionVersion]);
```

Comment:
so this usage of `useAsync` is what I also told Victorien - it's not really how it's intended.

`useAsync` itself already maintains the state for the result, in this case, `dmnParams`. The return value of the callback in `useAsync` is stored in the state. So, a more correct/intended usage is:

```suggestion
  const EMPTY_DMN_PARAMS = {inputs: [], outputs: []);

  const {loading, value: dmnParams = EMPTY_DMN_PARAMS} = useAsync(async () => {
    if (!pluginId || !decisionDefinitionId) {
      return EMPTY_DMN_PARAMS;
    }

    const queryParams = {
      engine: pluginId,
      definition: decisionDefinitionId,
    };

    if (decisionDefinitionVersion) {
      queryParams.version = decisionDefinitionVersion;
    }

    const response = await get(DMN_DECISION_DEFINITIONS_PARAMS_LIST, queryParams);
    const {inputs, outputs} = response.data;
    return {
      inputs: inputs.map(inputParam => [inputParam.expression, inputParam.label]),
      outputs: outputs.map(outputParam => [outputParam.name, outputParam.label]),
    };
  }, [pluginId, decisionDefinitionId, decisionDefinitionVersion]);
```